fineract-provider/src/main/resources/sql/migrations/core_db/V312__address_module_tables_metadat.sql

```diff
@@ -0,0 +1,142 @@
+	`postal_code` VARCHAR(10) NULL DEFAULT NULL,
+	`latitude` DECIMAL(10,8) UNSIGNED NULL DEFAULT '0.00000000',
+	`longitude` DECIMAL(10,8) UNSIGNED NULL DEFAULT '0.00000000',
+	`created_by` VARCHAR(100) NULL DEFAULT NULL,
+	`created_on` DATE NULL DEFAULT NULL,
+	`updated_by` VARCHAR(100) NULL DEFAULT NULL,
+	`updated_on` DATE NULL DEFAULT NULL,
+	PRIMARY KEY (`id`),
+	INDEX `address_fields_codefk1` (`state_province_id`),
+	INDEX `address_fields_codefk2` (`country_id`),
+	CONSTRAINT `address_fields_codefk1` FOREIGN KEY (`state_province_id`) REFERENCES `m_code_value` (`id`),
+	CONSTRAINT `address_fields_codefk2` FOREIGN KEY (`country_id`) REFERENCES `m_code_value` (`id`)
+)
+ENGINE=InnoDB
+AUTO_INCREMENT=1
+;
+
+
+-- entity address table
+CREATE TABLE `m_client_address` (
+	`id` BIGINT(20) NOT NULL AUTO_INCREMENT,
+	`client_id` BIGINT(20) NOT NULL DEFAULT '0',
+	`address_id` BIGINT(20) NOT NULL DEFAULT '0',
+	`address_type_id` INT(11) NOT NULL DEFAULT '0',
+	`is_active` TINYINT(4) NOT NULL DEFAULT '0',
+	PRIMARY KEY (`id`),
+	INDEX `addressIdFk` (`address_id`),
+	INDEX `address_codefk` (`address_type_id`),
+	INDEX `clientaddressfk` (`client_id`),
+	CONSTRAINT `address_codefk` FOREIGN KEY (`address_type_id`) REFERENCES `m_code_value` (`id`),
+	CONSTRAINT `clientaddressfk` FOREIGN KEY (`client_id`) REFERENCES `m_client` (`id`)
+)
+ENGINE=InnoDB
+AUTO_INCREMENT=1
+;
+
+
+-- field configuration
+CREATE TABLE `m_field_configuration` (
+	`id` BIGINT(20) NOT NULL AUTO_INCREMENT,
+	`entity` VARCHAR(100) NOT NULL,
+	`subentity` VARCHAR(100) NOT NULL,
+	`field` VARCHAR(100) NOT NULL,
+	`is_enabled` TINYINT(4) NOT NULL,
+	`is_mandatory` TINYINT(4) NOT NULL,
+	`validation_regex` VARCHAR(50) NULL DEFAULT NULL,
+	PRIMARY KEY (`id`)
+)
+ENGINE=InnoDB
+AUTO_INCREMENT=1
+;
+
+
+INSERT INTO `m_field_configuration` ( `entity`, `subentity`, `field`, `is_enabled`, `is_mandatory`, `validation_regex`) VALUES
+	('ADDRESS', 'CLIENT', 'addressType', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'street', 1, 1, ''),
+	('ADDRESS', 'CLIENT', 'addressLine1', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'addressLine2', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'addressLine3', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'townVillage', 0, 0, ''),
+	('ADDRESS', 'CLIENT', 'city', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'countyDistrict', 0, 0, ''),
+	('ADDRESS', 'CLIENT', 'stateProvinceId', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'countryId', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'postalCode', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'latitude', 0, 0, ''),
+	('ADDRESS', 'CLIENT', 'longitude', 0, 0, ''),
+	('ADDRESS', 'CLIENT', 'createdBy', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'createdOn', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'updatedBy', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'updatedOn', 1, 0, ''),
+	('ADDRESS', 'CLIENT', 'isActive', 1, 0, '');
+
+-- inserts for permission
+INSERT INTO `m_permission` ( `grouping`, `code`, `entity_name`, `action_name`, `can_maker_checker`) VALUES ('portfolio', 'CREATE_ADDRESS', 'ADDRESS', 'CREATE', 0);
```

Comment:
@nikpawar89  all the permissions are not covered like create, delete, update of address